cmd/up.go

```diff
@@ -356,7 +356,30 @@ func (up *UpContext) devMode(isRetry bool, d *appsv1.Deployment, create bool) (b
 		}
 	}
 
-	p, err := pods.GetDevPod(up.Context, up.Dev, up.Client, create)
+	reporter := make(chan string)
+	go func() {
+		long := false
+		message := ""
+		ticker := time.NewTicker(1000 * time.Millisecond)
+		for {
+			select {
+			case <-ticker.C:
+				if long && message != "" {
+					spinner.update(fmt.Sprintf("%s...", message))
+				}
+				long = true
+			case message = <-reporter:
+				if message == "" {
+					return
+				}
+				spinner.update("Activating your development environment...")
+				long = false
+			case <-up.Context.Done():
```

Comment:
Will this loop also exit when we close the `reporter` channel?

Reply:
I think this loop should depend on the lifecycle of `reporter`, and don't care about `up.Context.Done`. Instead, we close the reporter when done, and that can cause this loop to be closed.